compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2506,6 +2506,32 @@ public void visit(BLangForeach foreach) {
         result = blockNode;
     }
 
+    private BLangBlockStmt desugarForeachWithIteratorDef(BLangForeach foreach,
+                                                         BLangSimpleVariableDef dataVariableDefinition,
+                                                         BVarSymbol collectionSymbol,
+                                                         BInvokableSymbol iteratorInvokableSymbol,
+                                                         boolean isIteratorFuncFromLangLib) {
+        BLangBlockStmt blockNode;
+        BLangSimpleVariableDef iteratorVarDef = getIteratorVariableDefinition(foreach.pos, collectionSymbol,
+                iteratorInvokableSymbol, isIteratorFuncFromLangLib);
+        blockNode = desugarForeachToWhile(foreach, iteratorVarDef);
```

Comment:
Shall we inline declaration and initialization of blockNode
`BLangBlockStmt blockNode = desugarForeachToWhile(foreach, iteratorVarDef);`

Reply:
fixed